sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/CFBackedPageAsyncStream.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.android.communication.chat.implementation;
+
+import com.azure.android.core.logging.ClientLogger;
+import com.azure.android.core.rest.util.paging.PagedResponse;
+import com.azure.android.core.util.AsyncStream;
+import com.azure.android.core.util.AsyncStreamHandler;
+import com.azure.android.core.util.CancellationToken;
+import com.azure.android.core.util.Function;
+import com.azure.android.core.util.Predicate;
+
+import java.util.concurrent.CancellationException;
+
+import java9.util.concurrent.CompletableFuture;
+import java9.util.concurrent.CompletionException;
+
+public final class CFBackedPageAsyncStream<T>
+    implements AsyncStream<PagedResponse<T>> {
```

Comment:
This is just proof of concept, correct? Otherwise, move to com.azure.android.core somewhere.

Reply:
It's not proof of concept, the idea is this "base stream" will exist in all `CompletableFuture` based library as a "private" class (hopefully emitted by a hardcoded AutoRest template). This stream will back the core streams returned from public APIs.

Reply:
None of the core modules depends on any specific async library (CF, Rx, etc..), so if we want to move this to the core, it means we need a new core module for CF with only this class, so it's a trade of btw maintaining a new module vs having a private class in SDKs.

Reply:
Thanks for the explanation